src/test/java/com/tvd12/gamebox/testing/MMORoomGroupTest.java

```diff
@@ -123,4 +115,20 @@ public void destroyGroupTest() throws InterruptedException {
 				.collect(Collectors.toSet());
 		Asserts.assertFalse(threadNames.contains(roomGroupThreadName.getValue()));
 	}
+	
+	@Test
+	public void roomExceptionTest() throws InterruptedException {
+		// given
+		MMORoomGroup sut = MMORoomGroup.builder().timeTickMillies(100).build();
+		MMORoom room = mock(MMORoom.class);
+		doAnswer(invocation -> {
+			throw new Exception("Exception when room.update()");
+		}).when(room).update();
+		
+		// when
+		sut.addRoom(room);
+		Thread.sleep(300);
+		
+		// then
+	}
```

Comment:
Chỗ này em code như này sao nó không thow Exception anh nhỉ? Test exception thì mình nên verify như nào ở `//then` vậy anh?

Reply:
em định test trường hợp nó bắn ra exception và in ra log đúng không em?

Reply:
Vâng đúng rồi anh ạ

Reply:
Case này mình chỉ cần verify là có gọi qua `MMORoom.update` là được em ạ, còn log thì không cần em ạ, 1 phần vì không mock được logger, 1 phần là `MMORoom.update` đã chạy qua tức là case này đã hoạt động rồi em ạ, em dùng `doThrow ...` thay vì `doAnswer` nhé

Reply:
@tvd12 Anh review commit mới giúp em với ạ. Sao em chạy test không thấy có Exception trong log anh ạ.

Reply:
@vu-luong em thêm file `log4j.properties`

```
log4j.rootLogger=INFO, console

# For console
log4j.appender.console=org.apache.log4j.ConsoleAppender
log4j.appender.console.layout=org.apache.log4j.PatternLayout
log4j.appender.console.layout.ConversionPattern=%d{HH:mm:ss,SSS} %-5p [%t] %c{2} %3x - %m%n
```
vào folder `src/test/resources` em sẽ thấy log em nhé